package-res/resources/web/prompting/api/OperationAPI.js

```diff
@@ -48,39 +48,62 @@ define(['common-ui/prompting/PromptPanel', 'common-ui/prompting/parameters/Param
      *
      * @name OperationAPI#render
      * @method
-     * @param {String} id HTML object id where the render the panel
-     * @param {Function} getParameterXml The function called when the prompt panel needs to get the new parameter xml
+     * @param {String}    id HTML object id where the render the panel
+     * @param {Function}  getParameterXml The function called when the prompt panel needs to get the new parameter xml.
+     *                                    Receives a callback function. If synchronous, simply return xml. If asynchronous,
+     *                                    call the callback function provided and and pass xml to that.
+     * @example
+     *  // Asynchronous
+     *  api.operation.render("prompt-panel-render-area", function(api, callback) {
+     *    function retrieveXml(xml) {
+     *      callback(xml);
+     *    }
+     *  });
+     *
+     *  // Synchronous
+     *  api.operation.render("prompt-panel-render-area", function(api, callback) {
+     *    return xml; // return already retrieved xml definition
+     *  });
      */
     this.render = function(id, getParameterXml) {
       if (!getParameterXml || typeof getParameterXml != "function") {
         api.log.error(this._msgs.NO_PARAM_XML_FUNC, true);
       }
 
-      var xml = getParameterXml(api);
-      if (!xml) {
-        api.log.error(this._msgs.NO_PARAM_XML, true);
-      }
-
-      var paramDefn = this._parameterParser.parseParameterXml(xml);
-      this._promptPanel = new PromptPanel(id, paramDefn);
+      var getParameterXmlAsync = function(callback) {
+        var xml = getParameterXml(api, callback);
+        if (typeof xml != "undefined") {
```

Comment:
I check specifically for "undefined" here as an indication of whether we are trying to have a synchronous or asynchronous function call. if (!xml) will be for null, or other falsy values.